### PR TITLE
Add a NEWS entry covering the FIPS related changes.

### PR DESCRIPTION
**Pull request by paulidale**
Follow on to #21363

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by mattcaswell** [comment]
Shouldn't there be a CHANGES entry too?

**Comment by paulidale** [comment]
I didn't think we did both.

Easy enough to add something.

**Comment by mattcaswell** [comment]
> I didn't think we did both.

We do.

File: NEWS.md [hunk 1]
```diff
@@ -21,7 +21,12 @@ OpenSSL 3.1
 
 ### Major changes between OpenSSL 3.1.1 and OpenSSL 3.1.2 [under development]
 
-  * none
+  * By default, the FIPS provider is installed to ensure compliance in
```
Comment (beldmit) [inline NEWS.md:24]: to ensure FIPS-140-3 requirements as date of ...?

File: NEWS.md [hunk 2]
```diff
@@ -21,7 +21,12 @@ OpenSSL 3.1
...
+    its operation.  This installation has two notable side effects:
+
+    1. TLS 1.2 necessitates an extended master secret.  This could lead to
+       interoperability issues with older devices.
```
Comment (beldmit) [inline NEWS.md:28]: ...not supporting TLS 1.3 or EMS in TLS 1.2

**Review by beldmit — CHANGES_REQUESTED** [review]

**Comment by kroeckx** [comment]
If you don't know what that means, it's very confusing what that text says. I can read it as that the FIPS provider is now enabled by default.

File: NEWS.md (hunk 1 above)
Comment (mspncp) [reply NEWS.md:24]: or just

> to ensure FIPS-140-3 requirements

without a date?

Comment (beldmit) [reply NEWS.md:24]: As the requirements are a moving target, I think we should be somehow specific.

Comment (paulidale) [reply NEWS.md:24]: Both of these are in effect already.  Instead of a date, I'll add a reference to the relevant IG for each.

**Comment by paulidale** [comment]
Comments addressed.

File: CHANGES.md [hunk 3]
```diff
@@ -24,7 +24,12 @@ OpenSSL 3.1
 
 ### Changes between 3.1.1 and 3.1.2 [xx XXX xxxx]
 
- * none yet
+ * When building with the `enable-fips` option and using the resulting
+   FIPS provider, TLS 1.2 will mandate the use of an extended master
+   secret and the Hash and HMAC DRBGs will not operate with truncated
+   digests.
```
Comment (beldmit) [inline CHANGES.md:30]: This text contradicts the one from the NEWS - the NEWS one is mentioning default settings, and this one says about enforcing.

Comment (paulidale) [reply CHANGES.md:30]: _building with the `enable-fips` option and using the resulting FIPS provider_ is equivalent to my mind.

Comment (paulidale) [reply CHANGES.md:30]: The NEWS wording didn't seem appropriate to the CHANGES file.
NEWS is more verbose and explanatory IMO.

File: NEWS.md [hunk 4]
```diff
@@ -21,7 +21,17 @@ OpenSSL 3.1
...
+    installed to ensure FIPS 140-3 compliance by default.  This
+    installation has two notable side effects:
+
+    1. TLS 1.2 necessitates use of an extended master secret (EMS).
+       This could lead to interoperability issues with older devices
+       not supporting TLS 1.3 or EMS in TLS 1.2.  This is as per the
+       FIPS 140-3 IG G.Q transition.
+
+    2. Truncated digests cannot be utilised with the Hash and HMAC DRBGs.
+       This is as per the FIPS 140-3 IG G.Q transition.
```
Comment (t8m) [inline NEWS.md:34]: I would expect the NEWS.md entry being more terse than the CHANGES.md entry and not the other way around. This also sounds like just installation enables the FIPS provider where it is necessary to also modify the config file or load the provider from an application explicitly.

File: CHANGES.md (hunk 3 above)
Comment (t8m) [reply CHANGES.md:30]: > The NEWS wording didn't seem appropriate to the CHANGES file. NEWS is more verbose and explanatory IMO.

Nope? CHANGES.md entries are always the more verbose ones.

Comment (t8m) [reply CHANGES.md:30]: But IMO this text is fine and could go into both CHANGES.md and NEWS.md.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by paulidale** [comment]
Text updated.

File: NEWS.md [hunk 5]
```diff
@@ -21,7 +21,10 @@ OpenSSL 3.1
 
 ### Major changes between OpenSSL 3.1.1 and OpenSSL 3.1.2 [under development]
 
-  * none
+ * When building with the `enable-fips` option and using the resulting
+   FIPS provider, TLS 1.2 will mandate the use of an extended master
```
Comment (beldmit) [inline NEWS.md:25]: `will mandate` or `will mandate by default`?

Comment (paulidale) [reply NEWS.md:25]: updated.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Ping @openssl/committers for second review

**Review by tmshort — APPROVED** [review]
LGTM

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the feedback.

**Pull request closed without merging**